### ISSUE TITLE
Wordsearch: Prevent subset words generating

### ISSUE THREAD

**Issue by AndrewRigg** [AndrewRigg/Crossword#50]
Currently, words can exist which are subsets of other larger words.  This will not do!
(I had a wordsearch with WACKY and CAW which were using the same letters.)

**Comment by Golpette** [comment]
I need to check but I don't think this can happen - CAW would have existed somewhere else in the wordsearch

**Comment by AndrewRigg** [comment]
No, I'm pretty sure it did happen, it highlighted the same letters using my looping!  (If I see it again I will include a screenshot)

**Comment by Golpette** [comment]
Ok, just we fixed this in the crossword before... Let me know if you see it again and if you do it'll just take 5 mins for me to fix it 

(if i rememeber right it's where we check the ArrayList of Entry objects to set the "boolean seenBefore". before we just checked if words were equal but we should be checking if a previous word contains the new word...) 

maybe i did it wrong: maybe i made it so that it will only catch a smaller word that is already part of a bigger word, but if we have a small word perhaps we can place a larger word over the top and obviously the small word cannot .contain(bigger word)

**Comment by Golpette** [comment]
ok, so instead of typing that i should have just checked. I think you're right.

if FitWords_wordsearch.java, what do you think of changing

```
            if(word.toLowerCase().contains(entries.get(j).getWord().toLowerCase())){
                seenBefore = true;
            }
```

to

if(word.toLowerCase().contains(entries.get(j).getWord().toLowerCase())  ||
         entries.get(j).getWord().toLowerCase().contains( word.toLowerCase()  )     ){
    seenBefore = true;
}

i.e., we just need to check if new word is contained in any of the previous words _AND_ if any of the previous words are contained in the new word

**Comment by Golpette** [comment]
```
        if(word.toLowerCase().contains(entries.get(j).getWord().toLowerCase())  ||  
                      entries.get(j).getWord().toLowerCase().contains( word.toLowerCase() )    ){
            seenBefore = true;
        }
```

**Comment by Golpette** [comment]
naaaaa wait, im too tired. sorry, i dont think this could have happened like you said... we can't fit a word if there are no empty spaces... a new word could contain an old word but not the other way around

I'M GOING TO BED

**Comment by AndrewRigg** [comment]
![success](https://cloud.githubusercontent.com/assets/10067616/19413817/ae7bd79c-9331-11e6-84b3-2e7728c75ae0.PNG)

This was that screenshot from the other day evidencing that this is an issue.
